.github/workflows/gh-pages.yaml

```diff
@@ -3,36 +3,57 @@ name: CI/CD
 on: push
 
 env:
-  # Creates and uploads sourcemaps to Rollbar, and save the built extension as an artifact
   PUBLIC_RELEASE: ${{ github.ref == 'refs/heads/main' }}
 
 jobs:
-  build-stage:
+  build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v2
         with:
           node-version: '14'
+
+      #########################
+      # Prepare public folder #
+      #########################
+
+      - name: Create public distribution
+        run: |
+          mkdir -p public
+          mv example public
+        # Duplicate the line above to add more folders
+
+
+      #########
+      # Build # Each build should be self-contained
+      #########
+
+      - name: Build markdown files
+        run: |
+          npx markdown-styles@3.2.0 --layout github --input README.md --output public
+          mv public/README.html public/index.html
+
       - name: Build react-example
         working-directory: react-example
         env:
           PUBLIC_URL: "/playground/react-example"
-        run: yarn && yarn run build
+        run: yarn && yarn run build && mv build ../public/react-example
 
-      - name: Create public distribution
-        # Copy the directory without the directory name, see: https://unix.stackexchange.com/a/180987
-        run: |
-          mkdir public
-          mkdir public/example && cp --recursive example/. public/example
-          mkdir public/react-example && cp --recursive react-example/build/. public/react-example
+
+      ##########
+      # Deploy #
+      ##########
+
+      - uses: actions/upload-artifact@v2
+        if: ${{ ! fromJSON(env.PUBLIC_RELEASE)}}
```

Comment:
I restored the upload for the reason you mentioned before: it's good to see what the output is while testing a PR.

I just disabled it on `main`